tests/test_terminal_formatter.py

```
@@ -0,0 +1,435 @@
+            "privacy_tier": "local",
+            "analysis_duration_ms": 840,
+        },
+    }
+
+
+@pytest.mark.parametrize("schema_version", [None, "2", "v2", "2.0.0", 2])
+def test_terminal_formatter_rejects_malformed_schema_version(schema_version):
+    dg = _make_diffgraph()
+    dg["schema_version"] = schema_version
+
+    with pytest.raises(ValueError, match="MAJOR.MINOR"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)